### PR TITLE
`json-fortran` creates Fortran modules, so must be at compiler level

### PR DESCRIPTION
**Pull request by ocaisa**

### CODE REVIEW

**Comment by sassy-crick** [comment]
Test report by @sassy-crick
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
hp - Linux Debian GNU/Linux 11, x86_64, 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz, Python 3.9.2
See https://gist.github.com/sassy-crick/5d45f5d803b684571aad4b2e68ae7bb0 for a full test report.

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen2

File: easybuild/easyconfigs/m/mctc-lib/mctc-lib-0.3.1-GCC-11.3.0.eb [hunk 1]
```diff
@@ -22,7 +22,11 @@ builddependencies = [
     ('CMake', '3.23.1'),
 ]
 
-configopts = '-DBUILD_SHARED_LIBS=ON'
+dependencies = [
+    ('json-fortran', '8.3.0'),
+]
+
+configopts = ['-DBUILD_SHARED_LIBS=ON', '-DBUILD_SHARED_LIBS=OFF']
```
Comment (ocaisa) [inline easybuild/easyconfigs/m/mctc-lib/mctc-lib-0.3.1-GCC-11.3.0.eb:29]: This was inconsistent with the other esyconfigs so I fixed it

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen2

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso

**Review by jfgrimm — APPROVED** [review]
lgtm

**Comment by jfgrimm** [comment]
Going in, thanks @ocaisa!

**Pull request merged**